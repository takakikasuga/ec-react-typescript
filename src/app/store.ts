import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import userReducer from '../features/user/userSlice';
import itemsReducer from '../features/items/itemsSlice';
import itemsCountReducer from '../features/itemCount/itemCountSlice';
import itemPriceReducer from '../features/itemPrice/itemPriceSlice';
import addOrderReducer from '../features/order/orderSlice';
import orderUpdateReducer from '../features/order/orderUpdateSlice';
import fetchOrderReducer from '../features/order/fetchOrderSlice';
import cartListsReducer from '../features/cartLists/cartListsSlice';
import statusZeroIdReducer from '../features/statusZeroId/statusZeroIdSlice';
import updateOrderStatusReducer from '../features/order/updateOrderStatusSlice';
import orderHistoryReducer from '../features/order/orderHistorySlice';
import suggestItemsReducer from '../features/suggest/suggestSlice';
import adminReducer from '../features/admin/adminSlice';
import adminItemsReducer from '../features/items/adminItemsSlice';
import localCartStrageReducer from '../features/cartLists/localCartStrageSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    user: userReducer,
    items: itemsReducer,
    itemCount: itemsCountReducer,
    itemPrice: itemPriceReducer,
    addOrder: addOrderReducer,
    orderUpdate: orderUpdateReducer,
    fetchOrder: fetchOrderReducer,
    cartLists: cartListsReducer,
    statusZeroId: statusZeroIdReducer,
    updateOrderStatus: updateOrderStatusReducer,
    orderHistory: orderHistoryReducer,
    suggestItems: suggestItemsReducer,
    admin: adminReducer,
    adminItems: adminItemsReducer,
    localCartStrage: localCartStrageReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
