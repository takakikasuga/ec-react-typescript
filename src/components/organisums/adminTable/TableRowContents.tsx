import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

// コンポーネント
import { DeleteButton } from '../../atoms/button/DeleteButton'

// 型のインポート
import { FetchOrder } from "../../../types/order/order"

// 各種機能のインポート
import { deleteOrderItem, deleteOrderAsync, selectFetchOrder } from '../../../features/order/fetchOrderSlice'
import { selectUserId } from '../../../features/user/userSlice'
import { selectStatusZeroId } from '../../../features/statusZeroId/statusZeroIdSlice'
import { updateAdminItemsAsync } from "../../../features/items/adminItemsSlice"

// マテリアルUI
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import CardMedia from '@material-ui/core/CardMedia';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  media: {
    height: 200,
    width: 200
  },
  descriptionWidth: {
    maxWidth: "300px"
  }
});

// 現状propsの型定義方法の理解ができていない
export const TableRowContents = React.memo((props: any) => {
  const classes = useStyles();
  console.log(props)
  const { description, imagePath, name, price, uniqueId, id } = props
  const dipatch = useDispatch()
  // nullを「!」で明示的になくす
  const userId: string = useSelector(selectUserId)!
  const statusZeroId = useSelector(selectStatusZeroId)

  console.log("商品情報のユニークなID", uniqueId)

  // 注文情報の削除
  const deleteAdminItem = (uniqueId: string) => {
    // ユニークなIDを持たせてアップデートさせる
    if (window.confirm("本当に商品一覧から削除しますか？？")) {
      dipatch(updateAdminItemsAsync(uniqueId))
    }
  }
  return (
    <TableRow>
      <TableCell align="right">{id}</TableCell>
      <TableCell component="th" scope="row" style={{ width: '240px' }}>
        {name}
        <CardMedia
          className={classes.media}
          image={imagePath}
          title="Contemplative Reptile"
        />
      </TableCell>
      <TableCell className={classes.descriptionWidth} align="left">{description}</TableCell>
      <TableCell align="left">{name}</TableCell>
      <TableCell align="left">{(price.m)}</TableCell>
      <TableCell align="left">{(price.l)}</TableCell>
      <TableCell align="left">
        <span onClick={() => { deleteAdminItem(uniqueId) }}>
          <DeleteButton></DeleteButton>
        </span>
      </TableCell>
    </TableRow>
  )
})
