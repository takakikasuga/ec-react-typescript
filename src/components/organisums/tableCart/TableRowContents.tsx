import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

// コンポーネント
import { DeleteButton } from '../../atoms/button/DeleteButton'

// 型のインポート
import { FetchOrder } from "../../../types/order/order"

// 各種機能のインポート
import { deleteOrderItem, deleteOrderAsync, selectFetchOrder } from '../../../features/order/fetchOrderSlice'
import { selectUserId } from '../../../features/user/userSlice'
import { selectStatusZeroId } from '../../../features/statusZeroId/statusZeroIdSlice'
import { deleteLocalCartStrage } from "../../../features/cartLists/localCartStrageSlice"

// NoImageのインポート
import noImage from "../../../noImage/noImage.png"

// マテリアルUI
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import CardMedia from '@material-ui/core/CardMedia';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  media: {
    height: 200,
    width: 200
  },
});

// 現状propsの型定義方法の理解ができていない
export const TableRowContents = (props: any) => {
  const classes = useStyles();
  const { row, indexNum, imagePath } = props
  const dipatch = useDispatch()
  const fetchData: Array<FetchOrder> = useSelector(selectFetchOrder)
  // nullを「!」で明示的になくす
  const userId: string = useSelector(selectUserId)!
  const statusZeroId = useSelector(selectStatusZeroId)

  // 注文情報の削除
  const deleteCart = (indexNum: number) => {
    if (window.confirm("本当に削除しますか？？")) {
      // ログインしている状態であれば以下の処理を行う
      if (userId) {
        dipatch(deleteOrderItem(indexNum))
        // 直接参照しているstoreのデータを削除することができないのでコピー
        const updateFetchData = [...fetchData]
        // string型であることを保証する
        if (typeof userId === 'string') {
          updateFetchData.splice(indexNum, 1)
          dipatch(deleteOrderAsync({ userId, statusZeroId, updateFetchData }))
        }
        // ログインしている状態でなければ以下の処理を行う
      } else {
        dipatch(deleteLocalCartStrage(indexNum))
      }
    }
  }
  return (
    <TableRow>
      <TableCell component="th" scope="row" style={{ width: '240px' }}>
        {row.itemName}
        <CardMedia
          className={classes.media}
          // 管理者が消してしまった商品についてはnoImageを採用
          image={imagePath ? imagePath : noImage}
          title="Contemplative Reptile"
        />
      </TableCell>
      <TableCell align="right">{Number(row.itemPrice).toLocaleString()}円</TableCell>
      <TableCell align="right">{row.itemCount}個</TableCell>
      <TableCell align="right">{Number((row.itemPrice) * (row.itemCount)).toLocaleString()}円</TableCell>
      <TableCell align="right">
        <span onClick={() => { deleteCart(indexNum) }}>
          <DeleteButton></DeleteButton>
        </span>
      </TableCell>
    </TableRow>
  )
}
