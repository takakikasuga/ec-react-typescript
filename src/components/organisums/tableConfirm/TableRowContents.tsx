import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

// コンポーネント
import { DeleteButton } from '../../atoms/button/DeleteButton'
import noImage from "../../../noImage/noImage.png"

// 型のインポート
import { FetchOrder } from "../../../types/order/order"

// 各種機能のインポート
import { deleteOrderItem, deleteOrderAsync, selectFetchOrder } from '../../../features/order/fetchOrderSlice'
import { selectUserId } from '../../../features/user/userSlice'
import { selectStatusZeroId } from '../../../features/statusZeroId/statusZeroIdSlice'

// マテリアルUI
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import CardMedia from '@material-ui/core/CardMedia';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  media: {
    height: 200,
    width: 200
  },
});

// 現状propsの型定義方法の理解ができていない
export const TableRowContents = React.memo((props: any) => {
  const classes = useStyles();
  console.log(props)
  const { row, indexNum, imagePath } = props
  console.log(row, indexNum)
  const dipatch = useDispatch()
  const fetchData: Array<FetchOrder> = useSelector(selectFetchOrder)
  // nullを「!」で明示的になくす
  const userId: string = useSelector(selectUserId)!
  const statusZeroId = useSelector(selectStatusZeroId)


  // 注文情報の削除
  const deleteCart = (indexNum: number) => {
    dipatch(deleteOrderItem(indexNum))
    // 直接参照しているstoreのデータを削除することができないのでコピー
    const updateFetchData = [...fetchData]
    // string型であることを保証する
    if (typeof userId === 'string') {
      updateFetchData.splice(indexNum, 1)
      dipatch(deleteOrderAsync({ userId, statusZeroId, updateFetchData }))
    }
  }
  return (
    <TableRow>
      <TableCell component="th" scope="row" style={{ width: '240px' }}>
        {row.itemName}
        <CardMedia
          className={classes.media}
          // 商品情報が削除されていた場合のnoImage
          image={imagePath ? imagePath : noImage}
          title="Contemplative Reptile"
        />
      </TableCell>
      <TableCell align="right">{row.itemPrice}</TableCell>
      <TableCell align="right">{row.itemCount}</TableCell>
      <TableCell align="right">{((row.itemPrice) * (row.itemCount))}</TableCell>
    </TableRow>
  )
})
