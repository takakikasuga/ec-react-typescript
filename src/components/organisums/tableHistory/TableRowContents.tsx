import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

// コンポーネント
import noImage from "../../../noImage/noImage.png"

// 型のインポート
import { FetchOrder } from "../../../types/order/order"

// 各種機能のインポート
import { selectFetchOrder } from '../../../features/order/fetchOrderSlice'
import { selectUserId } from '../../../features/user/userSlice'
import { selectStatusZeroId } from '../../../features/statusZeroId/statusZeroIdSlice'
import { cancelOrderHistoryAsync } from '../../../features/order/orderHistorySlice'

// マテリアルUI
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import CardMedia from '@material-ui/core/CardMedia';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles({
  media: {
    height: 200,
    width: 200
  },
});

// 現状propsの型定義方法の理解ができていない
export const TableRowContents = React.memo((props: any) => {
  const classes = useStyles();
  console.log(props)
  const { orderList, indexNum, imagePath, order } = props
  console.log(orderList, indexNum)
  const dispatch = useDispatch()
  // nullを「!」で明示的になくす
  const userId: string = useSelector(selectUserId)!


  // 注文情報の削除のキャンセル
  const cancelOrder = (uniqueOrderId: string | undefined) => {
    // string型の保証
    if (typeof uniqueOrderId === "string") {
      if (window.confirm('本当にキャンセルしますか？')) {
        dispatch(cancelOrderHistoryAsync({ userId, uniqueOrderId }))
      }
    }

  }
  return (
    <TableRow  >
      <TableCell component="th" scope="row">
        {orderList.itemName}
        <CardMedia
          className={classes.media}
          image={imagePath ? imagePath : noImage}
          title="Contemplative Reptile"
        />
      </TableCell>
      <TableCell align="right">{orderList.itemPrice}</TableCell>
      <TableCell align="right">{orderList.itemCount}</TableCell>
      <TableCell align="right">{(orderList.itemPrice * orderList.itemCount).toLocaleString()}円（税込）</TableCell>
      {order.orderItems.length === (indexNum + 1)
        ? <TableCell align="right">{(order.totoalPrice).toLocaleString()}円（税込）</TableCell>
        : ""}
      {/* 発送済みまたはキャンセル済みまたはこの注文をキャンセルを場合分け */}
      <TableCell align="right">
        {order.orderItems.length === (indexNum + 1) && order.status !== 9
          ? <>
            <p>配達希望日：{order.orderDate}</p>
            <Button onClick={() => { cancelOrder(order.orderUniqueId) }} variant="contained">この注文をキャンセル</Button>

          </>
          : order.orderItems.length === (indexNum + 1) && order.status === 9
            ? <>
              <p>配達希望日：{order.orderDate}</p>
              <Button disabled={true} variant="contained">キャンセル済み</Button>
            </>
            : ""
        }
      </TableCell>
    </TableRow>
  )
})
