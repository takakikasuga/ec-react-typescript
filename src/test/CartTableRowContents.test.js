import React from "react"
import { render, cleanup, screen } from "@testing-library/react"
import { afterEach, expect } from "@jest/globals"
import { TableRowContents } from "../components/organisums/tableCart/TableRowContents"
import { store } from "../app/store"
import { Provider } from "react-redux"
import renderer from 'react-test-renderer';


describe("CartリストのTableRowContentsコンポーネント", () => {

  // ダミーデータ
  const row = {
    itemPrice: 3000,
    itemCount: 3,
  }

  // 各テストの直前に実行される（コンポーネントの取得）
  beforeEach(() => {
    render(
      <Provider store={store}>
        <TableRowContents row={row} />
      </Provider>
    )
  })

  // 各処理が通った後にアンマウントさせる
  afterEach(() => cleanup())

  describe("jest renderer test", () => {
    test("render correctly", () => {
      const tree = renderer
        .create(<Provider store={store}>
          <TableRowContents row={row} />
        </Provider>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  })
  describe("cartリストのrowコンテンツの確認", () => {
    test("rowコンテンツの金額が3000円であることの表示確認（パターン１）", () => {
      // screen.debug()
      screen.debug(screen.getByTestId("price"))
      // 完全一致パターン
      expect(screen.getByTestId("price")).toHaveTextContent(/^3,000円$/)
    })
    test("rowコンテンツの金額が3000円であることの表示確認（パターン２）", () => {
      // screen.debug()
      screen.debug(screen.getByTestId("price"))
      // textContentはDOM要素のtextContentにアクセス
      expect(screen.getByTestId("price").textContent).toBe("3,000円")
    })
    test("rowコンテンツの金額が3000円であることの表示確認（パターン３）", () => {
      // screen.debug()
      screen.debug(screen.getByTestId("price"))
      // .toBeInTheDocumentはDOM要素があるか否かの確認
      expect(screen.getByTestId("price")).toBeInTheDocument()
    })
    test("rowコンテンツの個数が3個であることの表示確認", () => {
      screen.debug()
      screen.debug(screen.getByTestId("count"))
      expect(screen.getByTestId("count").innerHTML).toBe("3個")
    })
    test("rowコンテンツに存在するtdタグの数をテスト", () => {
      expect(screen.getAllByRole("cell")).toHaveLength(5)
    })
  })
})