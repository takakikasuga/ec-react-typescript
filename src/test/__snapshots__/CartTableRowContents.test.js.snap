// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartリストのTableRowContentsコンポーネント jest renderer test render correctly 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <th
    aria-sort={null}
    className="MuiTableCell-root"
    role="cell"
    scope="row"
    style={
      Object {
        "width": "240px",
      }
    }
  >
    <div
      className="MuiCardMedia-root makeStyles-media-1"
      style={
        Object {
          "backgroundImage": "url(\\"noImage.png\\")",
        }
      }
      title="Contemplative Reptile"
    />
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
    data-testid="price"
  >
    3,000
    円
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
    data-testid="count"
  >
    3
    個
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    9,000
    円
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    <span
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          削除
        </span>
      </button>
    </span>
  </td>
</tr>
`;
